csrc/ops/arith.cpp

```diff
@@ -2617,4 +2634,67 @@ TensorView* prefixSum(TensorView* tv, int64_t dim) {
       /*init=*/tv->fusion()->zeroVal(tv->dtype()));
 }
 
+BlockQuantizationResults blockQuantize(TensorView* input) {
+  auto reshaped_input = reshape(input, [](auto& x) { x.split(-1, 16); });
+
+  auto inp_domain =
+      TensorDomain::noReductions(reshaped_input->getLogicalDomain());
+
+  // Validate input tensor is not zero-dimensional
+  NVF_CHECK(
+      !inp_domain.empty(),
+      "Block quantization does not support zero-dimensional tensors");
+
+  // Validate input data type - typically requires floating point input
+  NVF_CHECK(
+      isFloatingPointType(input->getDataType().value()),
+      "Block quantization expects floating point input but got ",
+      input->getDataType().value());
+
+  // Create output domain for quantized tensor (same shape as input)
+  std::vector<IterDomain*> quantized_out_domain;
+  quantized_out_domain.reserve(inp_domain.size());
+
+  for (auto inp_domain_ptr : inp_domain) {
+    quantized_out_domain.push_back(inp_domain_ptr->cloneWithoutRFactor());
+  }
+
+  // Create output domain for block scales
+  // Block scales typically have reduced dimensions based on block size
+  // For now, assuming block size of 16 and reducing the last dimension
+  std::vector<IterDomain*> scales_out_domain;
+  scales_out_domain.reserve(inp_domain.size());
+
+  for (size_t i = 0; i < inp_domain.size(); ++i) {
+    if (i == inp_domain.size() - 1) {
+      scales_out_domain.push_back(
```

Comment:
Do we need a broadcast here?

Reply:
I think we do -  it let's us schedule both the outputs in the same way.
Please let me know if that's not right.

Reply:
If we remove the reshape, I don't think it would mattter.

Reply:
Removed the reshape and the broadcast.